src/pages/outlier-detection/OutlierDetection.js

```diff
@@ -262,53 +262,53 @@ class OutlierDetection extends Page {
         this.setState({ algorithm: value })
     }
 
-    // toggleCheckbox(element) {
-    //     const api = this.context.d2.Api.getApi()
-    //     const elements = this.state.elements
-    //     for (let i = 0; i < elements.length; i++) {
-    //         const currentElement = elements[i]
-    //         if (currentElement.key === element.key) {
-    //             this.context.updateAppState({
-    //                 pageState: {
-    //                     loading: true,
-    //                 },
-    //             })
-    //             const apiCall = currentElement.marked ? api.delete : api.post
-    //             apiCall(
-    //                 apiConf.endpoints.markDataValue,
-    //                 OutlierAnalyisTable.convertElementToToggleFollowupRequest(
-    //                     currentElement
-    //                 )
-    //             )
-    //                 .then(() => {
-    //                     if (this.isPageMounted()) {
-    //                         currentElement.marked = !currentElement.marked
-    //                         elements[i] = currentElement
-
-    //                         this.context.updateAppState({
-    //                             showSnackbar: true,
-    //                             snackbarConf: {
-    //                                 type: SUCCESS,
-    //                                 message: i18n.t(
-    //                                     currentElement.marked
-    //                                         ? 'Marked'
-    //                                         : 'Unmarked'
-    //                                 ),
-    //                             },
-    //                             pageState: {
-    //                                 elements,
-    //                                 loading: false,
-    //                             },
-    //                         })
-    //                     }
-    //                 })
-    //                 .catch(error => {
-    //                     this.manageError(error)
-    //                 })
-    //             break
-    //         }
-    //     }
-    // }
+    toggleCheckbox(element) {
+        const api = this.context.d2.Api.getApi()
+        const elements = this.state.elements
+        for (let i = 0; i < elements.length; i++) {
+            const currentElement = elements[i]
+            if (currentElement.key === element.key) {
```

Comment:
This seems unnecessarily nested... what about changing the for-loop to a simple `find`?

```js
const elements = this.state.elements
const currentElement = elements.find(({ key }) => key === element.key)
if (!currentElement) {
   return
}
this.context.updateAppState({ // ... etc ...
```

Reply:
(if you need the index, use `findIndex` instead)

Reply:
The code was already there, but you are right that the nesting does not seem needed. I've adjusted the code in cdb6aff and have verified everything still works as expected.